src/WebJobs.Script/Host/ScriptHost.cs

```diff
@@ -313,15 +319,15 @@ public async Task InitializeAsync(CancellationToken cancellationToken = default)
 
                 var directTypes = GetDirectTypes(functionMetadataList);
                 await InitializeFunctionDescriptorsAsync(functionMetadataList, cancellationToken);
-
-                // Initialize worker function invocation dispatcher only for valid functions after creating function descriptors
-                // Dispatcher not needed for non-proxy codeless function.
-                // Disptacher needed for non-dotnet codeless functions
-                var filteredFunctionMetadata = functionMetadataList.Where(m => m.IsProxy() || !Utility.IsCodelessDotNetLanguageFunction(m));
-                await _functionDispatcher.InitializeAsync(Utility.GetValidFunctions(filteredFunctionMetadata, Functions), cancellationToken);
-
+                if (!workerIndexing)
```

Comment:
Can we check _functionDispatcher.State instead?

Reply:
workerIndexing variable also takes FeatureFlag into consideration.